tests/mir-opt/jump_threading.rs

```diff
@@ -88,7 +88,7 @@ fn two_reads() -> i32 {
     // CHECK:     [[c]] = copy [[a]];
     // CHECK:     [[tmp:_.*]] = copy [[c]];
     // CHECK:     [[eq:_.*]] = Eq(move [[tmp]], const 2_i32);
-    // CHECK:     switchInt(move [[eq]]) -> [0: bb2, otherwise: bb1];
+    // CHECK:     goto -> bb1;
```

Comment:
Thanks @saethlin for pushing me to investigate this. One of the behaviour changes is a bugfix for this pattern. On `master`, if read `a` twice, the `b = a` assignment makes us forget what the `c = a` assignment tells us. With this PR, we correctly merge the sets of conditions.